kate/recovery/src/commitments.rs

```diff
@@ -0,0 +1,107 @@
+use std::{array::TryFromSliceError, convert::TryFrom};
+
+use dusk_bytes::Serializable;
+use dusk_plonk::{
+	fft::{EvaluationDomain, Evaluations},
+	prelude::{BlsScalar, CommitKey, PublicParameters},
+};
+use thiserror::Error;
+
+use crate::com::{CHUNK_SIZE, COMMITMENT_SIZE};
+
+#[derive(Error, Debug)]
+pub enum Error {
+	#[error("Invalid data: {0}")]
+	InvalidData(String),
+	#[error("Cannot create {0}-bytes chunks from {1} bytes")]
+	InvalidChunksData(usize, usize),
+}
+
+impl From<TryFromSliceError> for Error {
+	fn from(e: TryFromSliceError) -> Self { Self::InvalidData(format!("{}", e)) }
+}
+
+impl From<dusk_bytes::Error> for Error {
+	fn from(e: dusk_bytes::Error) -> Self {
+		match e {
+			dusk_bytes::Error::InvalidData => {
+				Self::InvalidData("Scalar data is not valid".to_string())
+			},
+			dusk_bytes::Error::BadLength { .. } => {
+				Self::InvalidData("Invalid scalar data length".to_string())
+			},
+			dusk_bytes::Error::InvalidChar { .. } => {
+				Self::InvalidData("Scalar data contains invalid character".to_string())
+			},
+		}
+	}
+}
+
+impl From<dusk_plonk::error::Error> for Error {
+	fn from(e: dusk_plonk::error::Error) -> Self { Self::InvalidData(format!("{}", e)) }
+}
+
+fn try_into_scalar(chunk: &[u8]) -> Result<BlsScalar, Error> {
+	let sized_chunk = <[u8; CHUNK_SIZE]>::try_from(chunk)?;
+	BlsScalar::from_bytes(&sized_chunk).map_err(From::from)
+}
+
+fn try_into_scalars(data: &[u8]) -> Result<Vec<BlsScalar>, Error> {
+	let chunks = data.chunks_exact(CHUNK_SIZE);
+	if !chunks.remainder().is_empty() {
+		return Err(Error::InvalidData("Invalid data length".to_string()));
+	}
```

Comment:
Its nicer, but it couples this crate to substrate (which I tried to avoid). Does it make sense to depend on substrate only for this macro (or if there some other reason, I'm just guessing here)?